relayer/src/supervisor.rs

```diff
@@ -330,21 +335,42 @@ impl Supervisor {
         let mut subscriptions = Vec::with_capacity(self.config.chains.len());
 
         for chain_config in &self.config.chains {
-            let chain = self.registry.get_or_spawn(&chain_config.id)?;
-            let subscription = chain.subscribe()?;
-            subscriptions.push((chain, subscription));
+            let chain = match self.registry.get_or_spawn(&chain_config.id) {
+                Ok(chain) => chain,
+                Err(e) => {
+                    error!(
+                        "failed to spawn chain runtime for {}: {}",
+                        chain_config.id, e
+                    );
+                    continue;
+                }
+            };
+
+            match chain.subscribe() {
+                Ok(subscription) => subscriptions.push((chain, subscription)),
+                Err(e) => error!(
+                    "failed to subscribe to events of {}: {}",
+                    chain_config.id, e
+                ),
+            }
         }
 
-        self.spawn_workers()?;
+        self.spawn_workers();
 
         loop {
             match recv_multiple(&subscriptions) {
                 Ok((chain, batch)) => {
-                    self.process_batch(chain.clone(), batch.unwrap_or_clone())?;
-                }
-                Err(e) => {
-                    dbg!(e);
+                    let result = batch
+                        .unwrap_or_clone()
+                        .map_err(Into::into)
+                        .and_then(|batch| self.process_batch(chain.clone(), batch));
+
+                    match result {
+                        Ok(()) => trace!("[{}] batch processing complete", chain.id()),
```

Comment:
can we move this trace and we can figure it out later, we only want it for ibc events and not for new blocks. This is printed every block and even with 2 chains it makes the logs very verbose. I know it's tracing but still.

Reply:
Done